src/subs/diagmkl.f90

```diff
@@ -0,0 +1,43 @@
+SUBROUTINE DIAGMKL(NDIM,A,EVAL1,EVEC1,IERROR)
```

Comment:
Add copyrights with your information and describe what the subroutine does. See src/cuda/mgpu.h for an example.

Reply:
Added